cmd/oras/internal/display/status/progress/messenger.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package progress
+
+import (
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Messenger is progress message channel
+type Messenger struct {
+	ch chan *status
+}
+
+func (sm *Messenger) Start() {
+	if sm.ch == nil {
+		return
+	}
+	sm.ch <- startTiming()
+}
+
+func (sm *Messenger) Send(prompt string, descriptor ocispec.Descriptor, offset int64) {
+	for {
+		if sm.ch == nil {
+			return
+		}
+		select {
+		case sm.ch <- newStatusMessage(prompt, descriptor, offset):
+			return
+		case <-sm.ch:
+			// purge the channel until successfully pushed
+		}
+	}
+}
+
+func (sm *Messenger) Stop() {
+	if sm.ch == nil {
+		return
+	}
+	sm.ch <- endTiming()
+	close(sm.ch)
+	sm.ch = nil
```

Comment:
I would suggest to leave sm.ch there. Sending message to a closed status channel is a code-level defect and it's easier to be identified from panic(send to closed channel) to sleep(send to nil channel). Only when all goroutines are sleeping, there will be a panic.

Reply:
Done.